_posts/2020-09-15-newpipe-0.20.0.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: post
+title: "NewPipe 0.20.0 released: A new major version, a new unified player, a new notification experience... A lot is new this time!"
+short: "NewPipe 0.20.0 released"
+date: 2020-09-15 05:00:00 +0100
+categories: [pinned, announcement, release]
+author: Opusforlife & Poolitzer
+image: newpipe
+excerpt_separator: <!-- more -->
+---
+
+Hey there! Guess who is back? We are. Admittedly, we're quite a bit late with this release. But there is a _very_ good reason for that. Are you ready for it? Sit down, grab some popcorn and a beverage of your choice. This is going to be long.
+<!-- more -->
+
+## A new unified player
+
+"What in the name of the coronavirus is the _unified player_?" you ask. Well, first of all, that's a very weird expression. Where did you even get that one from... - you know what? Save it. We don't want to know. You just keep on using that, we'll keep on talking about the new player.
+
+A new player it indeed is. Up until now, NewPipe had three different players bundled in its code: the main, pop-up and background players. All three had separate code and separate playing queues. When you switched from one player to another, the stream would start from the beginning instead of carrying on from where you left. In the background, the old player and its queue closed itself, while the new one took over. If that sounds complicated and cumbersome, that's because it is. At least if you ask us.
```

Comment:
> When you switched from one player to another, the stream would start from the beginning instead of carrying on from where you left.

@Poolitzer Turns out this is only correct if you use the action buttons below the thumbnail. If you use the buttons in the player UI drop down, you correctly resume from where you left off (I think avently was the one who mentioned it in the RC thread). What to do here?

I'm thinking:

"All three had separate code and separate playing queues. When you switched from one player to another, in the background, the old player and its queue closed itself, while a new player was created, along with a new queue, after which it took over. If that sounds complicated and cumbersome, that's because it is. Moreover, if you accidentally switched using the player buttons below the thumbnail instead of the ones in the drop down menu in the player, the stream would start from the beginning instead of carrying on from where you left, and you would lose your (possibly carefully built) queue."

And for the next paras, something like:

"Well, no more! The buttons in the player drop down have been removed and the action buttons below the thumbnail work like they're supposed to, now, so you're at no risk whatsoever of losing your playing position or queue. Whew!

This was made possible by rewriting the code so that all players run on one common service. When you switch from one player to another, this service remains running, preserving the stream data, and the new player just picks up from where the old one left off. Everything remains the same, including the queue. Ah, serendipity."

Reply:
All three had separate code and separate playing queues. When you switched from one player to another, in the background, the old player and its queue closed itself, while a new player was created, along with a new queue, which took over then.

Sounds good?